src/typechecker.erl

```diff
@@ -709,11 +709,12 @@ normalize({type, _, record, [{atom, _, Name}|Fields]}, TEnv) when length(Fields)
         || ?type_field_type(FieldName, Type) <- Fields],
     type_record(Name, NormFields);
 normalize({type, _, union, Tys}, TEnv) ->
+    UnionSizeLimit = ?union_size_limit,
```

Comment:
Ah, now I see why we can't put it in Env: We only have TEnv here.

I think we should pass around the complete Env in many of these places where we only have Env, or use global options (persistent term or so) for all of them. I'm fine with this inconsistency for now, but it's better if we can have a more uniform way of doing options.